Fixing the error that arise when using uneven width and height

zeyuyun1 opened the pull request:
The current code for `SurfaceGeometry` works only when `width_segments==height_segments`. For example, 

```
surf_g = SurfaceGeometry(z=[0]*100, 
                         width_segments=nx - 1,
                         height_segments=ny - 1)
```
will only run when `nx == ny`.

Error arise when we have:

```
nx = 10
ny = 15
surf_g = SurfaceGeometry(z=[0]*150, 
                         width_segments=nx - 1,
                         height_segments=ny - 1)

ValueError: all the input array dimensions for the concatenation axis must match exactly, but along dimension 0, the array at index 0 has size 15 and the array at index 2 has size 10
```
This is because in [line 81](https://github.com/jupyter-widgets/pythreejs/blob/150ff1c10c868b17fefa63d19153b5ee1fe87f66/pythreejs/pythreejs.py#L81)
`z = np.array(self.z).reshape((nx, ny))` the reshaped dimension is reversed. 

The proposed change fixed this error without interrupting any other part.

zeyuyun1 committed:
Update pythreejs.py

fixing the error that arise when using uneven width and height.

vidartf commented:
Thanks for this contribution! I tested the fix locally using the code in examples/Examples.ipynb, and while this fixes the positions, it does seem like the indices are not correct in the nx != ny case. I'll dig some more on my side to see if I can spot the error quickly.

vidartf committed:
Fix grid indices

Uncovered this bug when testing new support for non-square sizes.

vidartf committed:
Fix non-square DataTexture

DataTexture seems to insist on column-major data. As arrays are always row-major (C-order) when serialized, we switch width/height on JS side. Some examples also need updating in order to highlight this, and to stay intuitive with respect to variable names.

Also regenerates modified example.

vidartf committed:
Fix CI deps

vidartf committed:
Use modern sphinx
